pkg/agent/clientset.go

```diff
@@ -218,15 +218,25 @@ func (cs *ClientSet) sync() {
 }
 
 func (cs *ClientSet) ServerCount() int {
-	var serverCount int
+	countFromLeases := 0
 	if cs.leaseCounter != nil {
-		serverCount = cs.leaseCounter.Count()
-	} else {
-		serverCount = cs.lastReceivedServerCount
+		countFromLeases = cs.leaseCounter.Count()
+	}
+	countFromResponses := cs.lastReceivedServerCount
+
+	serverCount := countFromLeases
+	countSource := "KNP server lease count"
+	if countFromResponses > serverCount {
+		serverCount = countFromResponses
```

Comment:
Not sure this is what we want. This clearly works when we are scaling up the number of KNP servers. However I believe this prevents us from easily scaling down the number of KNP servers. I thought was one of the goals of the lease system.

Reply:
That depends on how the static `--server-count` flag is set. With these changes it can:
  * be set to 1 to rely on leases alone
  * be set to some-default-value to provide a lower bound as a safety

The default lower bound is critical when we're gradually replacing agents and servers in parallel: 
We will still have a portion of the servers not recording their lease and a portion of agents assuming too-low count (because they only look at leases).